// @govuk-react/warning-text/dist/govuk-react-warning-text.cjs.prod.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var styled = require('styled-components');
var icons = require('@govuk-react/icons');
var lib = require('@govuk-react/lib');
var constants = require('@govuk-react/constants');

function _interopDefault (e) { return e && e.__esModule ? e : { 'default': e }; }

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n["default"] = e;
  return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);
var styled__default = /*#__PURE__*/_interopDefault(styled);

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};
  var target = _objectWithoutPropertiesLoose(source, excluded);
  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

var _excluded = ["children"];
var StyledWarningText = styled__default["default"]('div').withConfig({
  displayName: "src__StyledWarningText",
  componentId: "sc-14ghfbq-0"
})({
  alignItems: 'center',
  boxSizing: 'border-box',
  display: 'flex',
  width: '100%'
}, lib.spacing.withWhiteSpace({
  marginBottom: 0
}));
var IconImportantWrapper = styled__default["default"]('div').withConfig({
  displayName: "src__IconImportantWrapper",
  componentId: "sc-14ghfbq-1"
})({
  flex: 'none',
  height: 35,
  marginRight: constants.SPACING_POINTS[3],
  width: 35
});
var WarningTextWrapper = styled__default["default"]('strong').withConfig({
  displayName: "src__WarningTextWrapper",
  componentId: "sc-14ghfbq-2"
})(lib.typography.font({
  size: 19,
  weight: 'bold'
}));
/**
 * Use the warning text component when you need to warn users about something important, such as legal consequences of an action, or lack of action, that they might take.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/warning-text
 * - https://design-system.service.gov.uk/components/warning-text/
 */

var WarningText = function WarningText(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React__namespace.createElement(StyledWarningText, props, /*#__PURE__*/React__namespace.createElement(IconImportantWrapper, null, /*#__PURE__*/React__namespace.createElement(icons.IconImportant, null)), /*#__PURE__*/React__namespace.createElement(WarningTextWrapper, null, children));
};
WarningText.displayName = 'WarningText';

exports.WarningText = WarningText;
exports["default"] = WarningText;
