// @govuk-react/warning-text/dist/govuk-react-warning-text.esm.js
import * as React from 'react';
import styled from 'styled-components';
import { IconImportant } from '@govuk-react/icons';
import { spacing, typography } from '@govuk-react/lib';
import { SPACING_POINTS } from '@govuk-react/constants';

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};
  var target = _objectWithoutPropertiesLoose(source, excluded);
  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

var _excluded = ["children"];
var StyledWarningText = styled('div').withConfig({
  displayName: "src__StyledWarningText",
  componentId: "sc-14ghfbq-0"
})({
  alignItems: 'center',
  boxSizing: 'border-box',
  display: 'flex',
  width: '100%'
}, spacing.withWhiteSpace({
  marginBottom: 0
}));
var IconImportantWrapper = styled('div').withConfig({
  displayName: "src__IconImportantWrapper",
  componentId: "sc-14ghfbq-1"
})({
  flex: 'none',
  height: 35,
  marginRight: SPACING_POINTS[3],
  width: 35
});
var WarningTextWrapper = styled('strong').withConfig({
  displayName: "src__WarningTextWrapper",
  componentId: "sc-14ghfbq-2"
})(typography.font({
  size: 19,
  weight: 'bold'
}));
/**
 * Use the warning text component when you need to warn users about something important, such as legal consequences of an action, or lack of action, that they might take.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/warning-text
 * - https://design-system.service.gov.uk/components/warning-text/
 */

var WarningText = function WarningText(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React.createElement(StyledWarningText, props, /*#__PURE__*/React.createElement(IconImportantWrapper, null, /*#__PURE__*/React.createElement(IconImportant, null)), /*#__PURE__*/React.createElement(WarningTextWrapper, null, children));
};
WarningText.displayName = 'WarningText';

export { WarningText, WarningText as default };
