// @govuk-react/section-break/dist/govuk-react-section-break.cjs.dev.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var styled = require('styled-components');
var lib = require('@govuk-react/lib');
var govukColours = require('govuk-colours');

function _interopDefault (e) { return e && e.__esModule ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefault(styled);

/**
 * ### References
 *
 * - https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/core/_section-break.scss
 */
var breakSizes = {
  XL: 8,
  XLARGE: 8,
  L: 6,
  LARGE: 6,
  M: 4,
  MEDIUM: 4
};
/**
 * Create a thematic break between sections of content.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/section-break
 * - https://design-system.service.gov.uk/styles/typography/#section-break
 */

var SectionBreak = styled__default["default"]('hr').withConfig({
  displayName: "src__SectionBreak",
  componentId: "sc-fayw97-0"
})({
  margin: 0,
  border: 0
}, function (_ref) {
  var level = _ref.level;
  var size = breakSizes[level];

  if (size) {
    return lib.spacing.withWhiteSpace({
      margin: {
        size: size,
        direction: ['top', 'bottom']
      }
    });
  }

  return lib.spacing.withWhiteSpace();
}, function (_ref2) {
  var visible = _ref2.visible;
  return visible ? {
    borderBottom: "1px solid ".concat(govukColours.BORDER_COLOUR)
  } : undefined;
});
SectionBreak.defaultProps = {
  visible: undefined
};
SectionBreak.displayName = 'SectionBreak';

exports.SectionBreak = SectionBreak;
exports["default"] = SectionBreak;
