// @govuk-react/visually-hidden/dist/govuk-react-visually-hidden.esm.js
import styled from 'styled-components';
import { visuallyHidden } from '@govuk-react/lib';

/**
 * ### References
 *
 * - https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/helpers/_visually-hidden.scss
 * - https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/utilities/_visually-hidden.scss
 */
/**
 * Hide an element visually, but have it available for screen readers.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/visually-hidden
 */

var VisuallyHidden = styled('span').withConfig({
  shouldForwardProp: function shouldForwardProp(prop) {
    return !['important', 'focusable'].includes(prop);
  },
  displayName: "src__VisuallyHidden",
  componentId: "sc-nxwj55-0"
}).attrs(function (_ref) {
  var focusable = _ref.focusable,
      tabIndex = _ref.tabIndex;
  return (// if we're focusable but don't have a `tabIndex` set, add one
    focusable && tabIndex === undefined ? {
      tabIndex: '0'
    } : undefined
  );
})(function (_ref2) {
  var focusable = _ref2.focusable,
      important = _ref2.important;
  return visuallyHidden({
    focusable: focusable,
    important: important
  });
});
VisuallyHidden.defaultProps = {
  focusable: false,
  important: true
};
VisuallyHidden.displayName = 'VisuallyHidden';

export { VisuallyHidden, VisuallyHidden as default };
