// @govuk-react/panel/dist/govuk-react-panel.cjs.prod.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var styled = require('styled-components');
var govukColours = require('govuk-colours');
var lib = require('@govuk-react/lib');
var constants = require('@govuk-react/constants');
var polished = require('polished');

function _interopDefault (e) { return e && e.__esModule ? e : { 'default': e }; }

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n["default"] = e;
  return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);
var styled__default = /*#__PURE__*/_interopDefault(styled);

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};
  var target = _objectWithoutPropertiesLoose(source, excluded);
  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

var _excluded = ["title", "children"];

var _styled$withConfig;
var RAW_BORDER_WIDTH = Number(polished.stripUnit(constants.BORDER_WIDTH));
var StyledPanel = styled__default["default"]('div').withConfig({
  displayName: "src__StyledPanel",
  componentId: "sc-12tinr5-0"
})(lib.typography.font({
  size: 19
}), (_styled$withConfig = {
  boxSizing: 'border-box',
  marginBottom: constants.SPACING_POINTS[3],
  // NB govuk-frontend has this element styled tablet-first
  padding: constants.SPACING_POINTS[6] - RAW_BORDER_WIDTH,
  border: "".concat(constants.BORDER_WIDTH, " solid transparent"),
  textAlign: 'center'
}, _defineProperty(_styled$withConfig, constants.MEDIA_QUERIES.TABLET, {
  padding: constants.SPACING_POINTS[7] - RAW_BORDER_WIDTH
}), _defineProperty(_styled$withConfig, "color", govukColours.WHITE), _defineProperty(_styled$withConfig, "background", govukColours.TURQUOISE), _styled$withConfig), lib.spacing.withWhiteSpace()); // NB govuk-frontend allows for the headingLevel to change, but defaults to h1
// we're not supporting headingLevel at this time

var StyledTitle = styled__default["default"]('h1').withConfig({
  displayName: "src__StyledTitle",
  componentId: "sc-12tinr5-1"
})({
  marginTop: 0,
  marginBottom: constants.SPACING_POINTS[6],
  ':last-child': {
    marginBottom: 0
  }
}, lib.typography.font({
  size: 48,
  weight: 'bold'
}));
var StyledBody = styled__default["default"]('div').withConfig({
  displayName: "src__StyledBody",
  componentId: "sc-12tinr5-2"
})(lib.typography.font({
  size: 36
}));
/**
 * The panel component is a visible container used on confirmation or results pages to highlight important content.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/panel
 * - https://design-system.service.gov.uk/components/panel/
 *
 */

var Panel = function Panel(_ref) {
  var title = _ref.title,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React__namespace.createElement(StyledPanel, props, /*#__PURE__*/React__namespace.createElement(StyledTitle, null, title), children && /*#__PURE__*/React__namespace.createElement(StyledBody, null, children));
};
Panel.defaultProps = {
  children: undefined
};
Panel.displayName = 'Panel';

exports.Panel = Panel;
exports["default"] = Panel;
