// @govuk-react/table/dist/govuk-react-table.esm.js
import * as React from 'react';
import styled from 'styled-components';
import { typography, spacing } from '@govuk-react/lib';
import { FONT_WEIGHTS, SPACING_POINTS } from '@govuk-react/constants';
import { BORDER_COLOUR } from 'govuk-colours';

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};
  var target = _objectWithoutPropertiesLoose(source, excluded);
  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

// https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/table/_table.scss
var TableHeader = styled('thead').withConfig({
  displayName: "TableHeader",
  componentId: "sc-1n1nw4b-0"
})({});

// https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/table/_table.scss
var TableBody = styled('tbody').withConfig({
  displayName: "TableBody",
  componentId: "sc-1qqarm8-0"
})({});

// https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/table/_table.scss
var Caption = styled('caption').withConfig({
  displayName: "Caption",
  componentId: "sc-pvwfz5-0"
})({
  fontWeight: Number(FONT_WEIGHTS.bold),
  display: 'table-caption',
  textAlign: 'left'
});

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

var Cell = styled('td').withConfig({
  displayName: "Cell",
  componentId: "sc-msv0l6-0"
})(function (_ref) {
  var numeric = _ref.numeric,
      _ref$alignRight = _ref.alignRight,
      alignRight = _ref$alignRight === void 0 ? numeric : _ref$alignRight,
      bold = _ref.bold;
  return {
    padding: "".concat(SPACING_POINTS[2], "px ").concat(SPACING_POINTS[4], "px ").concat(SPACING_POINTS[2], "px 0"),
    borderBottom: "1px solid ".concat(BORDER_COLOUR),
    textAlign: alignRight ? 'right' : 'left',
    fontWeight: bold ? Number(FONT_WEIGHTS.bold) : undefined,
    ':last-child': {
      paddingRight: 0
    }
  };
}, function (_ref2) {
  var numeric = _ref2.numeric,
      isHeader = _ref2.isHeader;
  return numeric && !isHeader ? typography.font({
    tabular: true
  }) : undefined;
}, spacing.withWidth());

var CellHeader = function CellHeader(props) {
  return /*#__PURE__*/React.createElement(Cell, _extends({
    as: "th",
    bold: true,
    isHeader: true
  }, props));
};

// https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/table/_table.scss
var Row = styled('tr').withConfig({
  displayName: "Row",
  componentId: "sc-1y8eqfp-0"
})({});

var _excluded = ["caption", "children", "body", "head"];
var StyledTable = styled('table').withConfig({
  displayName: "src__StyledTable",
  componentId: "sc-16s660v-0"
})(typography.font({
  size: 19
}), typography.textColour, {
  width: '100%',
  borderSpacing: 0,
  borderCollapse: 'collapse'
}, spacing.withWhiteSpace({
  marginBottom: 6
}));
/**
 * Use the table component to make information easier to compare and scan for users.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/table
 * - https://design-system.service.gov.uk/components/table/
 */

var Table = function Table(_ref) {
  var caption = _ref.caption,
      children = _ref.children,
      _ref$body = _ref.body,
      body = _ref$body === void 0 ? children : _ref$body,
      head = _ref.head,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React.createElement(StyledTable, props, caption && /*#__PURE__*/React.createElement(Caption, null, caption), head && /*#__PURE__*/React.createElement(TableHeader, null, head), /*#__PURE__*/React.createElement(TableBody, null, body));
};
Table.defaultProps = {
  body: undefined,
  caption: undefined,
  children: undefined,
  head: undefined
};
Table.displayName = 'Table';
Table.CellHeader = CellHeader;
Table.Row = Row;
Table.Cell = Cell;
Table.Header = TableHeader;

export { Table, Table as default };
