// @govuk-react/section-break/dist/govuk-react-section-break.esm.js
import styled from 'styled-components';
import { spacing } from '@govuk-react/lib';
import { BORDER_COLOUR } from 'govuk-colours';

/**
 * ### References
 *
 * - https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/core/_section-break.scss
 */
var breakSizes = {
  XL: 8,
  XLARGE: 8,
  L: 6,
  LARGE: 6,
  M: 4,
  MEDIUM: 4
};
/**
 * Create a thematic break between sections of content.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/section-break
 * - https://design-system.service.gov.uk/styles/typography/#section-break
 */

var SectionBreak = styled('hr').withConfig({
  displayName: "src__SectionBreak",
  componentId: "sc-fayw97-0"
})({
  margin: 0,
  border: 0
}, function (_ref) {
  var level = _ref.level;
  var size = breakSizes[level];

  if (size) {
    return spacing.withWhiteSpace({
      margin: {
        size: size,
        direction: ['top', 'bottom']
      }
    });
  }

  return spacing.withWhiteSpace();
}, function (_ref2) {
  var visible = _ref2.visible;
  return visible ? {
    borderBottom: "1px solid ".concat(BORDER_COLOUR)
  } : undefined;
});
SectionBreak.defaultProps = {
  visible: undefined
};
SectionBreak.displayName = 'SectionBreak';

export { SectionBreak, SectionBreak as default };
