// @govuk-react/grid-row/dist/govuk-react-grid-row.cjs.dev.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var styled = require('styled-components');
var constants = require('@govuk-react/constants');
var lib = require('@govuk-react/lib');

function _interopDefault (e) { return e && e.__esModule ? e : { 'default': e }; }

var styled__default = /*#__PURE__*/_interopDefault(styled);

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

/**
 * Grid Row, to be used as a parent component for GridCol.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/grid-row
 */

var GridRow = styled__default["default"]('div').withConfig({
  displayName: "src__GridRow",
  componentId: "sc-1jwfkx4-0"
})(_defineProperty({
  display: 'flex',
  flexDirection: 'column',
  margin: "0 -".concat(constants.GUTTER_HALF)
}, constants.MEDIA_QUERIES.LARGESCREEN, {
  flexDirection: 'row'
}), lib.spacing.withWhiteSpace());
GridRow.displayName = 'GridRow';

exports.GridRow = GridRow;
exports["default"] = GridRow;
