// govuk-colours/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
// Based on https://github.com/alphagov/govuk-frontend/blob/master/src/globals/scss/vars/_colours.scss
var BLUE = exports.BLUE = "#1d70b8";
var MAINSTREAM_BRAND = exports.MAINSTREAM_BRAND = BLUE;
var PURPLE = exports.PURPLE = "#2e358b";
var PURPLE_50 = exports.PURPLE_50 = "#9799c4";
var PURPLE_25 = exports.PURPLE_25 = "#d5d6e7";
var MAUVE = exports.MAUVE = "#6f72af";
var MAUVE_50 = exports.MAUVE_50 = "#b7b9d7";
var MAUVE_25 = exports.MAUVE_25 = "#e2e2ef";
var FUSCHIA = exports.FUSCHIA = "#912b88";
var FUSCHIA_50 = exports.FUSCHIA_50 = "#c994c3";
var FUSCHIA_25 = exports.FUSCHIA_25 = "#e9d4e6";
var PINK = exports.PINK = "#d53880";
var PINK_50 = exports.PINK_50 = "#eb9bbe";
var PINK_25 = exports.PINK_25 = "#f6d7e5";
var BABY_PINK = exports.BABY_PINK = "#f499be";
var BABY_PINK_50 = exports.BABY_PINK_50 = "#faccdf";
var BABY_PINK_25 = exports.BABY_PINK_25 = "#fdebf2";
var RED = exports.RED = "#d4351c";
var RED_50 = exports.RED_50 = "#d9888c";
var RED_25 = exports.RED_25 = "#efcfd1";
var MELLOW_RED = exports.MELLOW_RED = "#df3034";
var MELLOW_RED_50 = exports.MELLOW_RED_50 = "#ef9998";
var MELLOW_RED_25 = exports.MELLOW_RED_25 = "#f9d6d6";
var ORANGE = exports.ORANGE = "#f47738";
var ORANGE_50 = exports.ORANGE_50 = "#fabb96";
var ORANGE_25 = exports.ORANGE_25 = "#fde4d4";
var BROWN = exports.BROWN = "#b58840";
var BROWN_50 = exports.BROWN_50 = "#dac39c";
var BROWN_25 = exports.BROWN_25 = "#f0e7d7";
var YELLOW = exports.YELLOW = "#ffdd00";
var YELLOW_50 = exports.YELLOW_50 = "#ffdf94";
var YELLOW_25 = exports.YELLOW_25 = "#fff2d3";
var GRASS_GREEN = exports.GRASS_GREEN = "#85994b";
var GRASS_GREEN_50 = exports.GRASS_GREEN_50 = "#c2cca3";
var GRASS_GREEN_25 = exports.GRASS_GREEN_25 = "#e7ebda";
var GREEN = exports.GREEN = "#006435";
var GREEN_50 = exports.GREEN_50 = "#7fb299";
var GREEN_25 = exports.GREEN_25 = "#cce0d6";
var TURQUOISE = exports.TURQUOISE = "#28a197";
var TURQUOISE_50 = exports.TURQUOISE_50 = "#95d0cb";
var TURQUOISE_25 = exports.TURQUOISE_25 = "#d5ecea";
var LIGHT_BLUE = exports.LIGHT_BLUE = "#2b8cc4";
var LIGHT_BLUE_50 = exports.LIGHT_BLUE_50 = "#96c6e2";
var LIGHT_BLUE_25 = exports.LIGHT_BLUE_25 = "#d5e8f3";
var BLACK = exports.BLACK = "#0b0c0c";
var GREY_1 = exports.GREY_1 = "#6f777b";
var GREY_2 = exports.GREY_2 = "#bfc1c3";
var GREY_3 = exports.GREY_3 = "#f3f2f1";
var GREY_4 = exports.GREY_4 = "#f8f8f8";
var WHITE = exports.WHITE = "#ffffff";
var LINK_COLOUR = exports.LINK_COLOUR = BLUE;
var LINK_ACTIVE_COLOUR = exports.LINK_ACTIVE_COLOUR = LIGHT_BLUE;
var LINK_HOVER_COLOUR = exports.LINK_HOVER_COLOUR = LIGHT_BLUE;
var LINK_VISITED_COLOUR = exports.LINK_VISITED_COLOUR = "#4c2c92";
var BUTTON_COLOUR = exports.BUTTON_COLOUR = "#00703c";
var BUTTON_COLOUR_DARKEN_15 = exports.BUTTON_COLOUR_DARKEN_15 = "#002413"; //darken(#00703c, 15%)
var FOCUS_COLOUR = exports.FOCUS_COLOUR = YELLOW;
var TEXT_COLOUR = exports.TEXT_COLOUR = BLACK;
var SECONDARY_TEXT_COLOUR = exports.SECONDARY_TEXT_COLOUR = GREY_1;
var BORDER_COLOUR = exports.BORDER_COLOUR = GREY_2;
var PANEL_COLOUR = exports.PANEL_COLOUR = GREY_3;
var CANVAS_COLOUR = exports.CANVAS_COLOUR = GREY_4;
var HIGHLIGHT_COLOUR = exports.HIGHLIGHT_COLOUR = GREY_4;
var PAGE_COLOUR = exports.PAGE_COLOUR = WHITE;
var DISCOVERY_COLOUR = exports.DISCOVERY_COLOUR = FUSCHIA;
var ALPHA_COLOUR = exports.ALPHA_COLOUR = PINK;
var BETA_COLOUR = exports.BETA_COLOUR = ORANGE;
var LIVE_COLOUR = exports.LIVE_COLOUR = GRASS_GREEN;
var BANNER_TEXT_COLOUR = exports.BANNER_TEXT_COLOUR = "#000000";
var ERROR_COLOUR = exports.ERROR_COLOUR = RED;
var ERROR_BACKGROUND = exports.ERROR_BACKGROUND = "#fef7f7";
var PROPOSITION_BORDER = exports.PROPOSITION_BORDER = "#2e3133";
var PROPOSITION_ACTIVE_NAV = exports.PROPOSITION_ACTIVE_NAV = "#1d8feb";
var FOOTER_BACKGROUND = exports.FOOTER_BACKGROUND = GREY_3;
var FOOTER_BORDER_TOP = exports.FOOTER_BORDER_TOP = "#a1acb2";
var FOOTER_LINK = exports.FOOTER_LINK = "#454a4c";
var FOOTER_LINK_HOVER = exports.FOOTER_LINK_HOVER = "#171819";
var FOOTER_TEXT = exports.FOOTER_TEXT = FOOTER_LINK;
var ATTORNEY_GENERALS_OFFICE = exports.ATTORNEY_GENERALS_OFFICE = "#9f1888";
var ATTORNEY_GENERALS_OFFICE_WEBSAFE = exports.ATTORNEY_GENERALS_OFFICE_WEBSAFE = "#a03a88";
var CABINET_OFFICE = exports.CABINET_OFFICE = "#005abb";
var CABINET_OFFICE_WEBSAFE = exports.CABINET_OFFICE_WEBSAFE = "#347da4";
var CIVIL_SERVICE = exports.CIVIL_SERVICE = "#af292e";
var DEPARTMENT_FOR_BUSINESS_INNOVATION_SKILLS = exports.DEPARTMENT_FOR_BUSINESS_INNOVATION_SKILLS = "#003479";
var DEPARTMENT_FOR_BUSINESS_INNOVATION_SKILLS_WEBSAFE = exports.DEPARTMENT_FOR_BUSINESS_INNOVATION_SKILLS_WEBSAFE = "#347da4";
var DEPARTMENT_FOR_COMMUNITIES_AND_LOCAL_GOVERNMENT = exports.DEPARTMENT_FOR_COMMUNITIES_AND_LOCAL_GOVERNMENT = "#00857e";
var DEPARTMENT_FOR_COMMUNITIES_AND_LOCAL_GOVERNMENT_WEBSAFE = exports.DEPARTMENT_FOR_COMMUNITIES_AND_LOCAL_GOVERNMENT_WEBSAFE = "#37836e";
var DEPARTMENT_FOR_CULTURE_MEDIA_SPORT = exports.DEPARTMENT_FOR_CULTURE_MEDIA_SPORT = "#d40072";
var DEPARTMENT_FOR_CULTURE_MEDIA_SPORT_WEBSAFE = exports.DEPARTMENT_FOR_CULTURE_MEDIA_SPORT_WEBSAFE = "#a03155";
var DEPARTMENT_FOR_EDUCATION = exports.DEPARTMENT_FOR_EDUCATION = "#003a69";
var DEPARTMENT_FOR_EDUCATION_WEBSAFE = exports.DEPARTMENT_FOR_EDUCATION_WEBSAFE = "#347ca9";
var DEPARTMENT_FOR_ENVIRONMENT_FOOD_RURAL_AFFAIRS = exports.DEPARTMENT_FOR_ENVIRONMENT_FOOD_RURAL_AFFAIRS = "#00a33b";
var DEPARTMENT_FOR_INTERNATIONAL_DEVELOPMENT = exports.DEPARTMENT_FOR_INTERNATIONAL_DEVELOPMENT = "#002878";
var DEPARTMENT_FOR_INTERNATIONAL_DEVELOPMENT_WEBSAFE = exports.DEPARTMENT_FOR_INTERNATIONAL_DEVELOPMENT_WEBSAFE = "#405e9a";
var DEPARTMENT_FOR_TRANSPORT = exports.DEPARTMENT_FOR_TRANSPORT = "#006c56";
var DEPARTMENT_FOR_TRANSPORT_WEBSAFE = exports.DEPARTMENT_FOR_TRANSPORT_WEBSAFE = "#398373";
var DEPARTMENT_FOR_WORK_PENSIONS = exports.DEPARTMENT_FOR_WORK_PENSIONS = "#00beb7";
var DEPARTMENT_FOR_WORK_PENSIONS_WEBSAFE = exports.DEPARTMENT_FOR_WORK_PENSIONS_WEBSAFE = "#37807b";
var DEPARTMENT_OF_ENERGY_CLIMATE_CHANGE = exports.DEPARTMENT_OF_ENERGY_CLIMATE_CHANGE = "#009ddb";
var DEPARTMENT_OF_ENERGY_CLIMATE_CHANGE_WEBSAFE = exports.DEPARTMENT_OF_ENERGY_CLIMATE_CHANGE_WEBSAFE = "#2b7cac";
var DEPARTMENT_OF_HEALTH = exports.DEPARTMENT_OF_HEALTH = "#00ad93";
var DEPARTMENT_OF_HEALTH_WEBSAFE = exports.DEPARTMENT_OF_HEALTH_WEBSAFE = "#39836e";
var FOREIGN_COMMONWEALTH_OFFICE = exports.FOREIGN_COMMONWEALTH_OFFICE = "#003e74";
var FOREIGN_COMMONWEALTH_OFFICE_WEBSAFE = exports.FOREIGN_COMMONWEALTH_OFFICE_WEBSAFE = "#406e97";
var GOVERNMENT_EQUALITIES_OFFICE = exports.GOVERNMENT_EQUALITIES_OFFICE = "#9325b2";
var HM_GOVERNMENT = exports.HM_GOVERNMENT = "#0076c0";
var HM_GOVERNMENT_WEBSAFE = exports.HM_GOVERNMENT_WEBSAFE = "#347da4";
var HM_REVENUE_CUSTOMS = exports.HM_REVENUE_CUSTOMS = "#009390";
var HM_REVENUE_CUSTOMS_WEBSAFE = exports.HM_REVENUE_CUSTOMS_WEBSAFE = "#008770";
var HM_TREASURY = exports.HM_TREASURY = "#af292e";
var HM_TREASURY_WEBSAFE = exports.HM_TREASURY_WEBSAFE = "#832322";
var HOME_OFFICE = exports.HOME_OFFICE = "#9325b2";
var HOME_OFFICE_WEBSAFE = exports.HOME_OFFICE_WEBSAFE = "#9440b2";
var MINISTRY_OF_DEFENCE = exports.MINISTRY_OF_DEFENCE = "#4d2942";
var MINISTRY_OF_DEFENCE_WEBSAFE = exports.MINISTRY_OF_DEFENCE_WEBSAFE = "#5a5c92";
var MINISTRY_OF_JUSTICE = exports.MINISTRY_OF_JUSTICE = "#231f20";
var MINISTRY_OF_JUSTICE_WEBSAFE = exports.MINISTRY_OF_JUSTICE_WEBSAFE = "#5a5c92";
var NORTHERN_IRELAND_OFFICE = exports.NORTHERN_IRELAND_OFFICE = "#002663";
var NORTHERN_IRELAND_OFFICE_WEBSAFE = exports.NORTHERN_IRELAND_OFFICE_WEBSAFE = "#3e598c";
var OFFICE_OF_THE_ADVOCATE_GENERAL_FOR_SCOTLAND = exports.OFFICE_OF_THE_ADVOCATE_GENERAL_FOR_SCOTLAND = "#002663";
var OFFICE_OF_THE_ADVOCATE_GENERAL_FOR_SCOTLAND_WEBSAFE = exports.OFFICE_OF_THE_ADVOCATE_GENERAL_FOR_SCOTLAND_WEBSAFE = LINK_COLOUR;
var OFFICE_OF_THE_LEADER_OF_THE_HOUSE_OF_LORDS = exports.OFFICE_OF_THE_LEADER_OF_THE_HOUSE_OF_LORDS = "#9c132e";
var OFFICE_OF_THE_LEADER_OF_THE_HOUSE_OF_LORDS_WEBSAFE = exports.OFFICE_OF_THE_LEADER_OF_THE_HOUSE_OF_LORDS_WEBSAFE = "#c2395d";
var SCOTLAND_OFFICE = exports.SCOTLAND_OFFICE = "#002663";
var SCOTLAND_OFFICE_WEBSAFE = exports.SCOTLAND_OFFICE_WEBSAFE = "#405c8a";
var THE_OFFICE_OF_THE_LEADER_OF_THE_HOUSE_OF_COMMONS = exports.THE_OFFICE_OF_THE_LEADER_OF_THE_HOUSE_OF_COMMONS = "#317023";
var THE_OFFICE_OF_THE_LEADER_OF_THE_HOUSE_OF_COMMONS_WEBSAFE = exports.THE_OFFICE_OF_THE_LEADER_OF_THE_HOUSE_OF_COMMONS_WEBSAFE = "#005f8f";
var UK_EXPORT_FINANCE = exports.UK_EXPORT_FINANCE = "#005747";
var UK_EXPORT_FINANCE_WEBSAFE = exports.UK_EXPORT_FINANCE_WEBSAFE = LINK_COLOUR;
var UK_TRADE_INVESTMENT = exports.UK_TRADE_INVESTMENT = "#c80651";
var UK_TRADE_INVESTMENT_WEBSAFE = exports.UK_TRADE_INVESTMENT_WEBSAFE = LINK_COLOUR;
var WALES_OFFICE = exports.WALES_OFFICE = "#a33038";
var WALES_OFFICE_WEBSAFE = exports.WALES_OFFICE_WEBSAFE = "#7a242a";
