// @govuk-react/heading/dist/govuk-react-heading.esm.js
import styled from 'styled-components';
import * as React from 'react';
import { HEADING_SIZES, TYPOGRAPHY_SCALE, MEDIA_QUERIES, LEVEL_TAG, LEVEL_SIZE } from '@govuk-react/constants';
import { typography, spacing } from '@govuk-react/lib';

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};
  var target = _objectWithoutPropertiesLoose(source, excluded);
  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

function ownKeys(object, enumerableOnly) {
  var keys = Object.keys(object);

  if (Object.getOwnPropertySymbols) {
    var symbols = Object.getOwnPropertySymbols(object);

    if (enumerableOnly) {
      symbols = symbols.filter(function (sym) {
        return Object.getOwnPropertyDescriptor(object, sym).enumerable;
      });
    }

    keys.push.apply(keys, symbols);
  }

  return keys;
}

function _objectSpread2(target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i] != null ? arguments[i] : {};

    if (i % 2) {
      ownKeys(Object(source), true).forEach(function (key) {
        _defineProperty(target, key, source[key]);
      });
    } else if (Object.getOwnPropertyDescriptors) {
      Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
      ownKeys(Object(source)).forEach(function (key) {
        Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
      });
    }
  }

  return target;
}

var _excluded = ["level", "size"];
// so if `size` is a string, we find a numeric size based off `HEADING_SIZES`
// but if `size` is a number we just send through that number

var StyledHeading = styled('h1').withConfig({
  displayName: "heading__StyledHeading",
  componentId: "sc-1dsobun-0"
})(typography.textColour, function (_ref) {
  var size = _ref.size;
  var actualSize = Number.isNaN(Number(size)) ? HEADING_SIZES[size] : size;

  if (!actualSize) {
    throw Error("Unknown size ".concat(size, " used for heading."));
  }

  return _objectSpread2({}, typography.font({
    size: actualSize,
    weight: 'bold'
  }));
}, {
  display: 'block',
  marginTop: 0
}, function (_ref2) {
  var size = _ref2.size;
  var actualSize = Number.isNaN(Number(size)) ? HEADING_SIZES[size] : size;
  var scaleInfo = TYPOGRAPHY_SCALE[actualSize];
  return _defineProperty({
    marginBottom: scaleInfo.mobile.spacing
  }, MEDIA_QUERIES.TABLET, {
    marginBottom: scaleInfo.tablet.spacing
  });
}, spacing.withWhiteSpace());
/**
 * Use heading tags, such as `<h1>`, `<h2>` and so on, to tag the headings on a page.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/heading
 * - https://design-system.service.gov.uk/styles/typography/#headings
 */

var Heading = function Heading(_ref4) {
  var level = _ref4.level,
      size = _ref4.size,
      props = _objectWithoutProperties(_ref4, _excluded);

  if (level) {
    if (process.env.NODE_ENV !== 'production') {
      // eslint-disable-next-line no-console
      console.warn("deprecated prop 'level' used in Heading, please replace with an \"H".concat(level, "\" component"));
    }

    if (LEVEL_TAG[level]) {
      return /*#__PURE__*/React.createElement(StyledHeading, _extends({
        size: LEVEL_SIZE[level]
      }, props, {
        as: LEVEL_TAG[level]
      }));
    }
  }

  return /*#__PURE__*/React.createElement(StyledHeading, _extends({
    size: size
  }, props));
};
Heading.defaultProps = {
  level: undefined,
  size: 'XLARGE',
  children: undefined
};
Heading.displayName = 'Heading';

var H1 = function H1(props) {
  return /*#__PURE__*/React.createElement(Heading, _extends({
    as: "h1",
    size: LEVEL_SIZE[1]
  }, props));
};
var H2 = function H2(props) {
  return /*#__PURE__*/React.createElement(Heading, _extends({
    as: "h2",
    size: LEVEL_SIZE[2]
  }, props));
};
var H3 = function H3(props) {
  return /*#__PURE__*/React.createElement(Heading, _extends({
    as: "h3",
    size: LEVEL_SIZE[3]
  }, props));
};
var H4 = function H4(props) {
  return /*#__PURE__*/React.createElement(Heading, _extends({
    as: "h4",
    size: LEVEL_SIZE[4]
  }, props));
};
var H5 = function H5(props) {
  return /*#__PURE__*/React.createElement(Heading, _extends({
    as: "h5",
    size: LEVEL_SIZE[5]
  }, props));
};
var H6 = function H6(props) {
  return /*#__PURE__*/React.createElement(Heading, _extends({
    as: "h6",
    size: LEVEL_SIZE[6]
  }, props));
};

export { H1, H2, H3, H4, H5, H6, Heading as default };
