// @govuk-react/pagination/dist/govuk-react-pagination.esm.js
import * as React from 'react';
import styled from 'styled-components';
import { NTA_LIGHT, FONT_SIZE, LINE_HEIGHT, MEDIA_QUERIES, SPACING, SPACING_POINTS } from '@govuk-react/constants';
import { typography, spacing } from '@govuk-react/lib';
import { BLUE, WHITE, GREY_4, YELLOW, PURPLE } from 'govuk-colours';
import { ArrowLeft, ArrowRight } from '@govuk-react/icons';

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};
  var target = _objectWithoutPropertiesLoose(source, excluded);
  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

var _excluded = ["previousPage", "nextPage", "children", "pageTitle"];

var _styled$withConfig;
var PaginationWrapper = styled('li').withConfig({
  displayName: "anchor__PaginationWrapper",
  componentId: "sc-2udijf-0"
})((_styled$withConfig = {
  boxSizing: 'border-box',
  margin: 0,
  padding: 0,
  listStyleType: 'none',
  fontFamily: NTA_LIGHT,
  fontWeight: 400,
  textTransform: 'none',
  fontSize: FONT_SIZE.SIZE_20,
  lineHeight: LINE_HEIGHT.SIZE_16,
  width: '100%'
}, _defineProperty(_styled$withConfig, MEDIA_QUERIES.LARGESCREEN, {
  fontSize: FONT_SIZE.SIZE_27,
  lineHeight: LINE_HEIGHT.SIZE_16
}), _defineProperty(_styled$withConfig, '> a', {
  boxSizing: 'border-box',
  color: BLUE,
  fill: 'currentColor',
  padding: SPACING.SCALE_3,
  backgroundColor: WHITE,
  textDecoration: 'none',
  outlineColor: 'transparent',
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  width: '100%',
  ':hover': {
    backgroundColor: GREY_4
  },
  ':focus': {
    outline: "3px solid ".concat(YELLOW)
  },
  ':visited': {
    color: PURPLE
  }
}), _styled$withConfig), function (_ref) {
  var previousPage = _ref.previousPage;
  return {
    marginRight: previousPage ? '3px' : undefined,
    ' > a': {
      alignItems: previousPage ? 'flex-start' : undefined
    },
    ' > a div': {
      justifyContent: previousPage ? 'flex-start' : undefined,
      ' > svg': {
        height: previousPage ? '13px' : undefined,
        width: previousPage ? '17px' : undefined,
        marginRight: previousPage ? '10px' : undefined
      }
    }
  };
}, function (_ref2) {
  var nextPage = _ref2.nextPage;
  return {
    ' > a': {
      alignItems: nextPage ? 'flex-end' : undefined
    },
    ' > a div': {
      justifyContent: nextPage ? 'flex-end' : undefined,
      ' > svg': {
        height: nextPage ? '13px' : undefined,
        width: nextPage ? '17px' : undefined,
        marginLeft: nextPage ? '10px' : undefined
      }
    }
  };
});
var InnerWrap = styled('div').withConfig({
  displayName: "anchor__InnerWrap",
  componentId: "sc-2udijf-1"
})({
  display: 'flex',
  alignItems: 'center',
  width: '100%'
});
var PageTitle = styled('span').withConfig({
  displayName: "anchor__PageTitle",
  componentId: "sc-2udijf-2"
})(_defineProperty({
  fontSize: FONT_SIZE.SIZE_14,
  lineHeight: LINE_HEIGHT.SIZE_14,
  textDecoration: 'underline'
}, MEDIA_QUERIES.LARGESCREEN, {
  fontSize: FONT_SIZE.SIZE_16,
  lineHeight: LINE_HEIGHT.SIZE_16
}));
var Anchor = styled('a').withConfig({
  displayName: "anchor__Anchor",
  componentId: "sc-2udijf-3"
})({});

var PaginationAnchor = function PaginationAnchor(_ref3) {
  var previousPage = _ref3.previousPage,
      nextPage = _ref3.nextPage,
      children = _ref3.children,
      pageTitle = _ref3.pageTitle,
      props = _objectWithoutProperties(_ref3, _excluded);

  return /*#__PURE__*/React.createElement(PaginationWrapper, {
    previousPage: previousPage,
    nextPage: nextPage
  }, /*#__PURE__*/React.createElement(Anchor, props, /*#__PURE__*/React.createElement(InnerWrap, null, previousPage && /*#__PURE__*/React.createElement(ArrowLeft, null), children, nextPage && /*#__PURE__*/React.createElement(ArrowRight, null)), pageTitle && /*#__PURE__*/React.createElement(PageTitle, null, pageTitle)));
};

PaginationAnchor.defaultProps = {
  previousPage: undefined,
  nextPage: undefined,
  pageTitle: undefined
};

var StyledList = styled('ul').withConfig({
  displayName: "src__StyledList",
  componentId: "sc-elluw3-0"
})(typography.font({
  size: 19
}), _defineProperty({
  boxSizing: 'border-box',
  display: 'flex',
  flexDirection: 'column',
  marginTop: SPACING_POINTS[5],
  padding: SPACING_POINTS[2],
  justifyContent: 'space-between',
  width: '100%'
}, MEDIA_QUERIES.LARGESCREEN, {
  flexDirection: 'row'
}), spacing.withWhiteSpace({
  marginBottom: 6
}));
/**
 * Navigational links that allow users to navigate within a series of pages or elements.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/pagination
 * - https://components.publishing.service.gov.uk/component-guide/previous_and_next_navigation
 *
 */

var Pagination = function Pagination(props) {
  return /*#__PURE__*/React.createElement(StyledList, props);
};
Pagination.displayName = 'Pagination';
Pagination.Anchor = PaginationAnchor;

export { Pagination, Pagination as default };
