// @govuk-react/ordered-list/dist/govuk-react-ordered-list.esm.js
import styled from 'styled-components';
import { SPACING_POINTS, MEDIA_QUERIES } from '@govuk-react/constants';
import { typography, spacing } from '@govuk-react/lib';
import ListItem from '@govuk-react/list-item';

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

function translateType(type) {
  return {
    bullet: 'disc',
    number: 'decimal'
  }[type] || type;
}

/**
 * Use lists to make blocks of text easier to read, and to break information into manageable chunks.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/ordered-list
 * - https://design-system.service.gov.uk/styles/typography/#lists
 */
var OrderedList = styled('ol').withConfig({
  displayName: "src__OrderedList",
  componentId: "sc-1392v8j-0"
})(typography.font({
  size: 19
}), typography.textColour, {
  marginTop: 0,
  '& &': {
    marginTop: SPACING_POINTS[2]
  }
}, spacing.withWhiteSpace({
  margin: {
    size: 4,
    direction: 'bottom'
  }
}), function (_ref) {
  var listStyleType = _ref.listStyleType;
  var type = translateType(listStyleType);
  return [{
    listStyleType: type
  }, type === 'none' ? {
    paddingLeft: 0
  } : {
    paddingLeft: SPACING_POINTS[4]
  }, // the inclusion of these means that withWhiteSpace's mb prop doesn't work on ListItem
  ['disc', 'decimal'].includes(type) ? _defineProperty({}, "> ".concat(ListItem), _defineProperty({
    marginBottom: 0
  }, MEDIA_QUERIES.TABLET, {
    marginBottom: SPACING_POINTS[1]
  })) : _defineProperty({}, "> ".concat(ListItem), {
    marginBottom: SPACING_POINTS[1]
  })];
});
OrderedList.defaultProps = {
  listStyleType: undefined
};
OrderedList.displayName = 'OrderedList'; // Export un-wrapped styled component

export { OrderedList, OrderedList as default };
