// @govuk-react/top-nav/dist/govuk-react-top-nav.cjs.dev.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var govukColours = require('govuk-colours');
var constants = require('@govuk-react/constants');
var lib = require('@govuk-react/lib');
var CrownIcon = require('@govuk-react/icon-crown');
var styled = require('styled-components');

function _interopDefault (e) { return e && e.__esModule ? e : { 'default': e }; }

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n["default"] = e;
  return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);
var CrownIcon__default = /*#__PURE__*/_interopDefault(CrownIcon);
var styled__default = /*#__PURE__*/_interopDefault(styled);

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};
  var target = _objectWithoutPropertiesLoose(source, excluded);
  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

function _classCallCheck(instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError("Cannot call a class as a function");
  }
}

function _defineProperties(target, props) {
  for (var i = 0; i < props.length; i++) {
    var descriptor = props[i];
    descriptor.enumerable = descriptor.enumerable || false;
    descriptor.configurable = true;
    if ("value" in descriptor) descriptor.writable = true;
    Object.defineProperty(target, descriptor.key, descriptor);
  }
}

function _createClass(Constructor, protoProps, staticProps) {
  if (protoProps) _defineProperties(Constructor.prototype, protoProps);
  if (staticProps) _defineProperties(Constructor, staticProps);
  return Constructor;
}

function _assertThisInitialized(self) {
  if (self === void 0) {
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  }

  return self;
}

function _setPrototypeOf(o, p) {
  _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) {
    o.__proto__ = p;
    return o;
  };

  return _setPrototypeOf(o, p);
}

function _inherits(subClass, superClass) {
  if (typeof superClass !== "function" && superClass !== null) {
    throw new TypeError("Super expression must either be null or a function");
  }

  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: {
      value: subClass,
      writable: true,
      configurable: true
    }
  });
  if (superClass) _setPrototypeOf(subClass, superClass);
}

function _getPrototypeOf(o) {
  _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) {
    return o.__proto__ || Object.getPrototypeOf(o);
  };
  return _getPrototypeOf(o);
}

function _isNativeReflectConstruct() {
  if (typeof Reflect === "undefined" || !Reflect.construct) return false;
  if (Reflect.construct.sham) return false;
  if (typeof Proxy === "function") return true;

  try {
    Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {}));
    return true;
  } catch (e) {
    return false;
  }
}

function _possibleConstructorReturn(self, call) {
  if (call && (typeof call === "object" || typeof call === "function")) {
    return call;
  } else if (call !== void 0) {
    throw new TypeError("Derived constructors may only return object or undefined");
  }

  return _assertThisInitialized(self);
}

function _createSuper(Derived) {
  var hasNativeReflectConstruct = _isNativeReflectConstruct();
  return function _createSuperInternal() {
    var Super = _getPrototypeOf(Derived),
        result;

    if (hasNativeReflectConstruct) {
      var NewTarget = _getPrototypeOf(this).constructor;
      result = Reflect.construct(Super, arguments, NewTarget);
    } else {
      result = Super.apply(this, arguments);
    }

    return _possibleConstructorReturn(this, result);
  };
}

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

var TopNavWrapper = styled__default["default"]('div').withConfig({
  displayName: "top-nav-wrapper__TopNavWrapper",
  componentId: "sc-1wnrl5k-0"
})(function (_ref) {
  var bgColor = _ref.bgColor,
      color = _ref.color;
  return _defineProperty({
    display: 'flex',
    justifyContent: 'center',
    backgroundColor: bgColor,
    color: color,
    fontFamily: constants.NTA_LIGHT,
    fontWeight: 400,
    fontSize: constants.FONT_SIZE.SIZE_14,
    lineHeight: constants.LINE_HEIGHT.SIZE_14
  }, constants.MEDIA_QUERIES.LARGESCREEN, {
    fontSize: constants.FONT_SIZE.SIZE_16,
    lineHeight: constants.LINE_HEIGHT.SIZE_16
  });
});

var TopNavInner = styled__default["default"]('div').withConfig({
  displayName: "top-nav-inner__TopNavInner",
  componentId: "sc-15yph1v-0"
})(_defineProperty({
  display: 'flex',
  flexDirection: 'column',
  width: "calc(100% - ".concat(constants.SPACING.SCALE_5, ")"),
  maxWidth: '960px',
  padding: "calc(".concat(constants.SPACING.SCALE_2, " * 0.8) ").concat(constants.SPACING.SCALE_3),
  boxSizing: 'border-box'
}, constants.MEDIA_QUERIES.LARGESCREEN, {
  padding: "calc(".concat(constants.SPACING.SCALE_2, " * 0.8) 0"),
  flexDirection: 'row',
  width: "calc(100% - ".concat(constants.SPACING.SCALE_6, ")")
}));

var BottomNavWrapper = styled__default["default"]('div').withConfig({
  displayName: "bottom-nav-wrapper__BottomNavWrapper",
  componentId: "sc-1334g24-0"
})(_defineProperty({
  borderBottom: "10px solid ".concat(govukColours.BLUE),
  maxWidth: '960px',
  margin: '0 auto',
  width: "calc(100% - ".concat(constants.SPACING.SCALE_5, ")")
}, constants.MEDIA_QUERIES.LARGESCREEN, {
  width: "calc(100% - ".concat(constants.SPACING.SCALE_6, ")")
}));

var Company = styled__default["default"]('div').withConfig({
  displayName: "company__Company",
  componentId: "sc-fxukuq-0"
})({
  display: 'flex',
  fontWeight: 800,
  alignItems: 'center',
  justifyContent: 'flex-start',
  fontSize: '30px'
});

var LogoSearchWrapper = styled__default["default"]('div').withConfig({
  displayName: "logo-search-wrapper__LogoSearchWrapper",
  componentId: "sc-1iq7ssu-0"
})(_defineProperty({
  display: 'flex',
  justifyContent: 'space-between'
}, constants.MEDIA_QUERIES.LARGESCREEN, {
  display: 'block',
  paddingRight: constants.SPACING.SCALE_3,
  width: '33.33%'
}));

var RightHandSide = styled__default["default"]('div').withConfig({
  displayName: "right-hand-side__RightHandSide",
  componentId: "sc-1c11aed-0"
})(_defineProperty({
  display: 'flex',
  flexDirection: 'row',
  flexWrap: 'wrap',
  justifyContent: 'space-between',
  marginTop: constants.SPACING.SCALE_2
}, constants.MEDIA_QUERIES.LARGESCREEN, {
  flexDirection: 'column',
  alignItems: 'flex-start',
  justifyContent: 'flex-start',
  width: '66.66%',
  paddingLeft: constants.SPACING.SCALE_3,
  marginTop: 0
}));

var SearchWrapper = styled__default["default"]('div').withConfig({
  displayName: "search-wrapper__SearchWrapper",
  componentId: "sc-on0jnt-0"
})(_defineProperty({
  marginTop: constants.SPACING.SCALE_1,
  width: '50%',
  minWidth: '110px'
}, constants.MEDIA_QUERIES.LARGESCREEN, {
  width: '100%'
}));

var Ul = styled__default["default"]('ul').withConfig({
  shouldForwardProp: function shouldForwardProp(prop) {
    return !['serviceTitle', 'open'].includes(String(prop));
  },
  displayName: "unordered-list__Ul",
  componentId: "sc-otbf2p-0"
})(function (_ref) {
  var serviceTitle = _ref.serviceTitle,
      open = _ref.open;
  return _defineProperty({
    display: open ? 'flex' : 'none',
    flexWrap: 'wrap',
    margin: 0,
    padding: 0,
    marginTop: constants.SPACING.SCALE_1,
    width: '100%'
  }, constants.MEDIA_QUERIES.LARGESCREEN, {
    display: 'flex',
    paddingTop: serviceTitle ? 0 : constants.SPACING.SCALE_5,
    width: 'auto'
  });
});

var Li = styled__default["default"]('li').withConfig({
  displayName: "list-item__Li",
  componentId: "sc-1kicpn1-0"
})(_defineProperty({
  flex: '1 0 50%',
  width: '100%',
  listStyleType: 'none',
  margin: 0,
  padding: '3px 0',
  borderBottom: "1px solid ".concat(govukColours.PROPOSITION_BORDER)
}, constants.MEDIA_QUERIES.LARGESCREEN, {
  borderBottom: 0,
  flex: 'none',
  width: 'auto',
  paddingRight: constants.SPACING.SCALE_3
}));

// in order to provide toggle function of mobile menu
// when javascript is not present.

var Button = styled__default["default"]('label').withConfig({
  displayName: "menu-button__Button",
  componentId: "sc-1yd67mz-0"
})(_defineProperty({
  display: 'flex',
  alignItems: 'center'
}, constants.MEDIA_QUERIES.LARGESCREEN, {
  display: 'none'
}));
var ButtonText = styled__default["default"]('div').withConfig({
  displayName: "menu-button__ButtonText",
  componentId: "sc-1yd67mz-1"
})({
  cursor: 'default',
  ':hover': {
    textDecoration: 'underline'
  }
});
var ButtonIcon = styled__default["default"]('div').withConfig({
  displayName: "menu-button__ButtonIcon",
  componentId: "sc-1yd67mz-2"
})(function (_ref) {
  var open = _ref.open;
  return {
    content: '""',
    display: 'inline-block',
    width: '0',
    height: '0',
    borderStyle: 'solid',
    borderColor: 'transparent',
    clipPath: open ? 'polygon(50% 0%, 0% 100%, 100% 100%)' : 'polygon(0% 0%, 50% 100%, 100% 0%)',
    borderWidth: open ? '0 5px 8.66px 5px' : '8.66px 5px 0 5px',
    marginLeft: '5px',
    borderTopColor: 'inherit',
    borderBottomColor: 'inherit'
  };
});

var MenuButton = function MenuButton(_ref2) {
  var title = _ref2.title,
      open = _ref2.open,
      onClick = _ref2.onClick;
  return /*#__PURE__*/React__namespace["default"].createElement(Button, {
    onClick: onClick,
    htmlFor: "govuk-react-menu-button"
  }, /*#__PURE__*/React__namespace["default"].createElement(ButtonText, null, title), /*#__PURE__*/React__namespace["default"].createElement(ButtonIcon, {
    open: open
  }));
};

MenuButton.defaultProps = {
  title: 'Menu',
  open: false,
  onClick: function onClick() {
    return null;
  }
};

var Container = styled__default["default"]('div').withConfig({
  displayName: "icon-title__Container",
  componentId: "sc-onvli0-0"
})({
  verticalAlign: 'baseline'
});
var IconContainer = styled__default["default"]('span').withConfig({
  displayName: "icon-title__IconContainer",
  componentId: "sc-onvli0-1"
})({
  width: '36px',
  maxHeight: '27px',
  display: 'inline-block',
  marginRight: '7px'
});
var TitleContainer = styled__default["default"]('span').withConfig({
  displayName: "icon-title__TitleContainer",
  componentId: "sc-onvli0-2"
})({
  display: 'inline-block'
});

var IconTitle = function IconTitle(_ref) {
  var icon = _ref.icon,
      children = _ref.children;
  return /*#__PURE__*/React__namespace.createElement(Container, null, /*#__PURE__*/React__namespace.createElement(IconContainer, null, icon), /*#__PURE__*/React__namespace.createElement(TitleContainer, null, children));
};

IconTitle.defaultProps = {
  icon: undefined,
  children: undefined
};
IconTitle.displayName = 'TopNav.IconTitle';

var TopNavAnchor = styled__default["default"]('a').withConfig({
  displayName: "top-nav-anchor__TopNavAnchor",
  componentId: "sc-lt39h5-0"
})({
  color: govukColours.WHITE,
  textDecoration: 'none',
  textDecorationSkipInk: 'none',
  borderBottom: '1px solid transparent',
  fontWeight: 700,
  lineHeight: 1,
  ':hover': {
    borderBottomColor: govukColours.WHITE
  },
  ':focus': {
    outline: "3px solid ".concat(govukColours.YELLOW)
  }
});
TopNavAnchor.displayName = 'TopNav.Anchor';

var NavLinkAnchor = styled__default["default"](TopNavAnchor).withConfig({
  displayName: "nav-link-anchor__NavLinkAnchor",
  componentId: "sc-1pr115z-0"
})({
  display: 'inline-block',
  ':focus': {
    color: govukColours.BLACK,
    backgroundColor: govukColours.YELLOW,
    outline: "3px solid ".concat(govukColours.YELLOW),
    ':hover': {
      borderBottomColor: govukColours.BLACK
    }
  }
});
NavLinkAnchor.displayName = 'TopNav.NavLink';

var _excluded = ["bgColor", "color", "company", "serviceTitle", "search", "children"];

var ServiceTitleWrapper = styled__default["default"]('div').withConfig({
  displayName: "src__ServiceTitleWrapper",
  componentId: "sc-140rlix-0"
})(lib.typography.font({
  size: 24
}), _defineProperty({
  width: '50%'
}, constants.MEDIA_QUERIES.LARGESCREEN, {
  width: 'auto'
})); // Layout/position of MenuButtonWrapper

var MenuButtonWrapper = styled__default["default"]('div').withConfig({
  displayName: "src__MenuButtonWrapper",
  componentId: "sc-140rlix-1"
})(_defineProperty({
  display: 'flex',
  justifyContent: 'flex-end',
  alignItems: 'flex-end',
  width: '50%'
}, constants.MEDIA_QUERIES.LARGESCREEN, {
  width: 'auto'
})); // Provides a fallback for the mobile menu toggle
// functionality when javascrcipt is unavailable

var Input = styled__default["default"]('input').withConfig({
  displayName: "src__Input",
  componentId: "sc-140rlix-2"
})(_defineProperty({
  position: 'absolute',
  top: '-999em',
  ':checked + ul': {
    display: 'flex'
  }
}, constants.MEDIA_QUERIES.LARGESCREEN, {
  display: 'none'
}));

/**
 * The GOV.UK header shows users that they are on GOV.UK and which service they are using.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/top-nav
 * - https://design-system.service.gov.uk/components/header/
 */
var TopNav = /*#__PURE__*/function (_Component) {
  _inherits(TopNav, _Component);

  var _super = _createSuper(TopNav);

  function TopNav(props) {
    var _this;

    _classCallCheck(this, TopNav);

    _this = _super.call(this, props);

    _defineProperty(_assertThisInitialized(_this), "toggleNavigationOpen", function () {
      _this.setState(function (prevState) {
        return {
          navigationOpen: !prevState.navigationOpen
        };
      });
    });

    var defaultOpen = _this.props.defaultOpen;
    _this.state = {
      navigationOpen: defaultOpen
    };
    return _this;
  }

  _createClass(TopNav, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          bgColor = _this$props.bgColor,
          color = _this$props.color,
          company = _this$props.company,
          serviceTitle = _this$props.serviceTitle,
          search = _this$props.search,
          children = _this$props.children,
          props = _objectWithoutProperties(_this$props, _excluded);

      var navigationOpen = this.state.navigationOpen;
      return /*#__PURE__*/React__namespace["default"].createElement(React__namespace["default"].Fragment, null, /*#__PURE__*/React__namespace["default"].createElement(TopNavWrapper, _extends({
        bgColor: bgColor,
        color: color
      }, props), /*#__PURE__*/React__namespace["default"].createElement(TopNavInner, null, /*#__PURE__*/React__namespace["default"].createElement(LogoSearchWrapper, null, /*#__PURE__*/React__namespace["default"].createElement(Company, null, company), search && /*#__PURE__*/React__namespace["default"].createElement(SearchWrapper, null, search)), /*#__PURE__*/React__namespace["default"].createElement(RightHandSide, null, /*#__PURE__*/React__namespace["default"].createElement(ServiceTitleWrapper, null, serviceTitle), children && /*#__PURE__*/React__namespace["default"].createElement(React__namespace["default"].Fragment, null, /*#__PURE__*/React__namespace["default"].createElement(MenuButtonWrapper, null, /*#__PURE__*/React__namespace["default"].createElement(MenuButton, {
        open: navigationOpen
      })), /*#__PURE__*/React__namespace["default"].createElement(Input, {
        id: "govuk-react-menu-button",
        type: "checkbox",
        checked: navigationOpen,
        onChange: this.toggleNavigationOpen
      }), /*#__PURE__*/React__namespace["default"].createElement(Ul, {
        id: "govuk-react-menu",
        serviceTitle: !!serviceTitle,
        open: navigationOpen
      }, Array.isArray(children) ? React__namespace["default"].Children.map(children, function (child) {
        return child ? /*#__PURE__*/React__namespace["default"].createElement(Li, null, child) : null;
      }) : /*#__PURE__*/React__namespace["default"].createElement(Li, null, children)))))), /*#__PURE__*/React__namespace["default"].createElement(BottomNavWrapper, null));
    }
  }]);

  return TopNav;
}(React.Component);

_defineProperty(TopNav, "IconTitle", IconTitle);

_defineProperty(TopNav, "Anchor", TopNavAnchor);

_defineProperty(TopNav, "NavLink", NavLinkAnchor);

_defineProperty(TopNav, "displayName", 'TopNav');

_defineProperty(TopNav, "defaultProps", {
  bgColor: govukColours.BLACK,
  color: govukColours.WHITE,
  company: /*#__PURE__*/React__namespace["default"].createElement(IconTitle, {
    icon: /*#__PURE__*/React__namespace["default"].createElement(CrownIcon__default["default"], {
      width: "36",
      height: "32"
    })
  }, "GOV.UK"),
  serviceTitle: undefined,
  search: false,
  children: undefined,
  defaultOpen: false
});

exports.TopNav = TopNav;
exports["default"] = TopNav;
