// @govuk-react/tabs/dist/govuk-react-tabs.esm.js
import * as React from 'react';
import styled from 'styled-components';
import { spacing, typography, link } from '@govuk-react/lib';
import { BORDER_COLOUR, BLACK, WHITE, GREY_4 } from 'govuk-colours';
import { MEDIA_QUERIES } from '@govuk-react/constants';

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

var TabList = styled('ul').withConfig({
  displayName: "list__TabList",
  componentId: "sc-srtuwr-0"
})({
  margin: 0
}, spacing.responsive({
  size: 6,
  property: 'margin',
  direction: 'bottom'
}), _defineProperty({
  padding: 0,
  listStyle: 'none'
}, MEDIA_QUERIES.TABLET, {
  borderBottom: "1px solid ".concat(BORDER_COLOUR),
  marginBottom: 0,
  '::after': {
    content: "''",
    display: 'block',
    clear: 'both'
  }
}));

var TabPanel = styled('section').withConfig({
  displayName: "panel__TabPanel",
  componentId: "sc-1c8lkp4-0"
})(spacing.responsiveMargin({
  size: 8,
  direction: 'bottom'
}), _defineProperty({}, MEDIA_QUERIES.TABLET, {
  marginBottom: spacing.simple(0),
  paddingTop: spacing.simple(6),
  paddingRight: spacing.simple(4),
  paddingBottom: spacing.simple(6),
  paddingLeft: spacing.simple(4),
  border: "1px solid ".concat(BORDER_COLOUR),
  borderTop: 0,
  '& > :last-child': {
    marginBottom: 0
  }
}), function (_ref) {
  var selected = _ref.selected;
  return _defineProperty({
    display: 'block'
  }, MEDIA_QUERIES.TABLET, {
    display: !selected && 'none'
  });
});

var spacingSimple1 = spacing.simple(1);
var spacingSimple2 = spacing.simple(2);
var spacingSimple3 = spacing.simple(3);
var spacingSimple4 = spacing.simple(4);
var spacingSimple5 = spacing.simple(5);
var StyledListItem = styled('li').withConfig({
  displayName: "tab__StyledListItem",
  componentId: "sc-1e33m29-0"
})(_defineProperty({
  marginLeft: spacingSimple5,
  ':before': {
    content: "'\\2014  '",
    marginLeft: -spacingSimple5,
    paddingRight: spacingSimple1
  }
}, MEDIA_QUERIES.TABLET, {
  marginLeft: 0,
  ':before': {
    content: 'none'
  }
}));
var StyledHyperLink = styled('a').withConfig({
  displayName: "tab__StyledHyperLink",
  componentId: "sc-1e33m29-1"
})(typography.font({
  size: 19
}), link.common(), link.styleDefault, {
  display: 'inline-block',
  paddingTop: spacingSimple2,
  paddingBottom: spacingSimple2
}, function (_ref) {
  var selected = _ref.selected;
  return _defineProperty({
    color: selected && BLACK,
    textDecoration: selected ? 'none' : undefined
  }, MEDIA_QUERIES.TABLET, {
    marginRight: spacingSimple1,
    "float": 'left',
    textAlign: 'center',
    textDecoration: 'none',
    color: BLACK,
    ':link,:visited': {
      color: BLACK
    },
    marginTop: selected ? -spacingSimple1 : undefined,
    marginBottom: selected ? -1 : undefined,
    paddingTop: selected ? spacingSimple3 - 1 : undefined,
    paddingRight: selected ? spacingSimple4 - 1 : spacingSimple4,
    paddingBottom: selected ? spacingSimple3 + 1 : undefined,
    paddingLeft: selected ? spacingSimple4 - 1 : spacingSimple4,
    border: selected && "1px solid ".concat(BORDER_COLOUR),
    borderBottom: selected ? 0 : 1,
    backgroundColor: selected ? WHITE : GREY_4,
    ':focus': {
      backgroundColor: selected ? 'transparent' : GREY_4
    }
  });
});

var Tab = function Tab(props) {
  return /*#__PURE__*/React.createElement(StyledListItem, null, /*#__PURE__*/React.createElement(StyledHyperLink, props));
};

Tab.defaultProps = {
  selected: false
};

var TabsTitle = styled('h2').withConfig({
  displayName: "title__TabsTitle",
  componentId: "sc-5azs4d-0"
})(typography.font({
  size: 19
}), _defineProperty({
  marginBottom: spacing.simple(1)
}, MEDIA_QUERIES.TABLET, {
  display: 'none'
}));
TabsTitle.defaultProps = {
  children: 'Contents'
};

/**
 * ### References:
 *
 * - https://github.com/alphagov/govuk-frontend/tree/main/src/govuk/components/tabs
 */
var TabsContainer = styled('div').withConfig({
  displayName: "src__TabsContainer",
  componentId: "sc-k251l1-0"
})(typography.font({
  size: 19
}), typography.textColour, spacing.responsiveMargin({
  size: 1,
  direction: 'top'
}), spacing.responsiveMargin({
  size: 6,
  direction: 'bottom'
}), spacing.withWhiteSpace());
/**
 * The tabs component lets users navigate between related sections of content, displaying one section at a time.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/tabs
 * - https://design-system.service.gov.uk/components/tabs/
 */

var Tabs = function Tabs(props) {
  return /*#__PURE__*/React.createElement(TabsContainer, props);
};
Tabs.displayName = 'Tabs';
Tabs.Tab = Tab;
Tabs.List = TabList;
Tabs.Panel = TabPanel;
Tabs.Title = TabsTitle;

export { TabList as List, TabPanel as Panel, Tab, Tabs, TabsTitle as Title, Tabs as default };
