// @govuk-react/constants/dist/govuk-react-constants.esm.js
import { FOCUS_COLOUR } from 'govuk-colours';

// Tracking:
var FOCUS_WIDTH = '3px';
var FOCUS_WIDTH_RAW = 3;
var FOCUSABLE = {
  '&:focus': {
    outline: "".concat(FOCUS_WIDTH, " solid ").concat(FOCUS_COLOUR),
    outlineOffset: 0
  }
};
var FOCUSABLE_FILL = {
  '&:focus': {
    outline: "".concat(FOCUS_WIDTH, " solid ").concat(FOCUS_COLOUR),
    outlineOffset: 0,
    backgroundColor: FOCUS_COLOUR
  }
};

// https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/settings/_spacing.scss
var RESPONSIVE_0 = {
  mobile: 0,
  tablet: 0
};
var RESPONSIVE_1 = {
  mobile: 5,
  tablet: 5
};
var RESPONSIVE_2 = {
  mobile: 10,
  tablet: 10
};
var RESPONSIVE_3 = {
  mobile: 15,
  tablet: 15
};
var RESPONSIVE_4 = {
  mobile: 15,
  tablet: 20
};
var RESPONSIVE_5 = {
  mobile: 15,
  tablet: 25
};
var RESPONSIVE_6 = {
  mobile: 20,
  tablet: 30
};
var RESPONSIVE_7 = {
  mobile: 25,
  tablet: 40
};
var RESPONSIVE_8 = {
  mobile: 30,
  tablet: 50
};
var RESPONSIVE_9 = {
  mobile: 40,
  tablet: 60
};
var SPACING_MAP = [RESPONSIVE_0, RESPONSIVE_1, RESPONSIVE_2, RESPONSIVE_3, RESPONSIVE_4, RESPONSIVE_5, RESPONSIVE_6, RESPONSIVE_7, RESPONSIVE_8, RESPONSIVE_9];
var SPACING_MAP_INDEX = Array.from(Array(SPACING_MAP.length).keys());
var SPACING_POINTS = {
  0: 0,
  1: 5,
  2: 10,
  3: 15,
  4: 20,
  5: 25,
  6: 30,
  7: 40,
  8: 50,
  9: 60
}; // Ref: https://github.com/alphagov/govuk-frontend/blob/68bd09bb3e54b7ef4b4084ad2b3336858923a041/src/settings/_measurements.scss#L23

var WIDTHS = {
  'one-quarter': '25%',
  'one-third': '33.3333%',
  'one-half': '50%',
  'two-thirds': '66.6666%',
  'three-quarters': '75%',
  full: '100%'
};

// https://design-system.service.gov.uk/styles/typography/
var TYPOGRAPHY_SCALE = {
  80: {
    mobile: {
      fontSize: 53,
      lineHeight: 55 / 53,
      spacing: RESPONSIVE_8.mobile
    },
    tablet: {
      fontSize: 80,
      lineHeight: 1,
      // 80 / 80
      spacing: RESPONSIVE_8.tablet
    },
    print: {
      fontSize: 53,
      lineHeight: 1.1
    }
  },
  48: {
    mobile: {
      fontSize: 32,
      lineHeight: 35 / 32,
      spacing: RESPONSIVE_8.mobile
    },
    tablet: {
      fontSize: 48,
      lineHeight: 50 / 48,
      spacing: RESPONSIVE_8.tablet
    },
    print: {
      fontSize: 32,
      lineHeight: 1.15
    }
  },
  36: {
    mobile: {
      fontSize: 24,
      lineHeight: 25 / 24,
      spacing: RESPONSIVE_6.mobile
    },
    tablet: {
      fontSize: 36,
      lineHeight: 40 / 36,
      spacing: RESPONSIVE_6.tablet
    },
    print: {
      fontSize: 24,
      lineHeight: 1.05
    }
  },
  27: {
    mobile: {
      fontSize: 18,
      lineHeight: 20 / 18,
      spacing: RESPONSIVE_4.mobile
    },
    tablet: {
      fontSize: 27,
      lineHeight: 30 / 27,
      spacing: RESPONSIVE_4.tablet
    },
    print: {
      fontSize: 18,
      lineHeight: 1.15
    }
  },
  24: {
    mobile: {
      fontSize: 18,
      lineHeight: 20 / 18,
      spacing: RESPONSIVE_4.mobile
    },
    tablet: {
      fontSize: 24,
      lineHeight: 30 / 24,
      spacing: RESPONSIVE_4.tablet
    },
    print: {
      fontSize: 18,
      lineHeight: 1.15
    }
  },
  19: {
    mobile: {
      fontSize: 16,
      lineHeight: 20 / 16,
      spacing: RESPONSIVE_4.mobile
    },
    tablet: {
      fontSize: 19,
      lineHeight: 25 / 19,
      spacing: RESPONSIVE_4.tablet
    },
    print: {
      fontSize: 14,
      lineHeight: 1.15
    }
  },
  16: {
    mobile: {
      fontSize: 14,
      lineHeight: 16 / 14,
      spacing: RESPONSIVE_4.mobile
    },
    tablet: {
      fontSize: 16,
      lineHeight: 20 / 16,
      spacing: RESPONSIVE_4.tablet
    },
    print: {
      fontSize: 14,
      lineHeight: 1.2
    }
  },
  14: {
    mobile: {
      fontSize: 12,
      lineHeight: 15 / 12,
      spacing: RESPONSIVE_4.mobile
    },
    tablet: {
      fontSize: 14,
      lineHeight: 20 / 14,
      spacing: RESPONSIVE_4.tablet
    },
    print: {
      fontSize: 12,
      lineHeight: 1.2
    }
  }
}; // NB Spacing values set above relate to headings
// body classes use different responsive margins
// captions use govuk-spacing(1) for xl and l sizes only
// heading sizes supported in govuk-frontend

var HEADING_SIZES = {
  XLARGE: 48,
  XL: 48,
  LARGE: 36,
  L: 36,
  MEDIUM: 24,
  M: 24,
  SMALL: 19,
  S: 19
}; // caption sizes supported in govuk-frontend

var CAPTION_SIZES = {
  XLARGE: 27,
  XL: 27,
  LARGE: 24,
  L: 24,
  MEDIUM: 19,
  M: 19
}; // body text sizes supported in govuk-frontend

var BODY_SIZES = {
  LARGE: 24,
  L: 24,
  MEDIUM: 19,
  M: 19,
  SMALL: 16,
  S: 16,
  XSMALL: 14,
  XS: 14
}; // Lookup between numerical header level and associated html element

var LEVEL_TAG = {
  1: 'h1',
  2: 'h2',
  3: 'h3',
  4: 'h4',
  5: 'h5',
  6: 'h6'
}; // Lookup between a header level (1-6) and the associated default font size

var LEVEL_SIZE = {
  1: 'XLARGE',
  2: 'LARGE',
  3: 'MEDIUM',
  4: 'SMALL',
  5: 'SMALL',
  6: 'SMALL'
};
var FONT_WEIGHTS = {
  bold: '700',
  regular: '400'
};

var BORDER_WIDTH = '5px';
var BORDER_WIDTH_WIDE = '10px';
var BORDER_WIDTH_MOBILE = '4px';
var BORDER_WIDTH_FORM_ELEMENT = '2px';
var BORDER_WIDTH_FORM_ELEMENT_ERROR = '4px';
var BORDER_WIDTH_FORM_GROUP_ERROR = BORDER_WIDTH;
var SITE_WIDTH_PX = 960;
var SITE_WIDTH = "".concat(SITE_WIDTH_PX, "px");
var FULL_WIDTH = '100%'; // tracks $govuk-breakpoints at https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/settings/_media-queries.scss

var BREAKPOINTS = {
  SMALLSCREEN: '320px',
  // @deprecated
  LARGESCREEN: '641px',
  // @deprecated
  MOBILE: '320px',
  TABLET: '641px',
  DESKTOP: '769px'
};
var GUTTER_PX = 30;
var GUTTER = "".concat(GUTTER_PX, "px");
var GUTTER_HALF = "".concat(GUTTER_PX / 2, "px");
var GUTTER_ONE_THIRD = "".concat(GUTTER_PX / 3, "px");
var SPACING = {
  SCALE_1: '5px',
  SCALE_2: '10px',
  SCALE_3: '15px',
  SCALE_4: '20px',
  SCALE_5: '30px',
  SCALE_6: '60px'
};
var SITE_WIDTH_PLUS_GUTTERS = "".concat(SITE_WIDTH_PX + 2 * GUTTER_PX, "px");
var MEDIA_QUERY_PRINT = '@media print';
var MEDIA_QUERIES = {
  LARGESCREEN: "@media only screen and (min-width: ".concat(BREAKPOINTS.LARGESCREEN, ")"),
  MAX: "@media only screen and (min-width: ".concat(SITE_WIDTH_PLUS_GUTTERS, ")"),
  PRINT: MEDIA_QUERY_PRINT,
  TABLET: "@media only screen and (min-width: ".concat(BREAKPOINTS.TABLET, ")"),
  DESKTOP: "@media only screen and (min-width: ".concat(BREAKPOINTS.DESKTOP, ")")
};
var NTA_LIGHT = '"nta", Arial, sans-serif';
var NTA_LIGHT_TABULAR = "\"ntatabularnumbers\", ".concat(NTA_LIGHT);
var FONT_STACK = "".concat(NTA_LIGHT);
var FONT_STACK_TABULAR = "".concat(NTA_LIGHT_TABULAR);
var FONT_STACK_PRINT = 'sans-serif';
var FONT_SIZE = {
  SIZE_14: '14px',
  SIZE_16: '16px',
  SIZE_18: '18px',
  // not directly used in govuk-frontend
  SIZE_19: '19px',
  SIZE_20: '20px',
  // Not used in govuk-frontend
  SIZE_24: '24px',
  SIZE_27: '27px'
};
var LINE_HEIGHT = {
  SIZE_14: '1.1428571429',
  SIZE_16: '1.25',
  SIZE_18: '1.3',
  // approximate - value in typography is correct
  SIZE_19: '1.3',
  // ditto
  // SIZE_24 differs here from 24 def in typography scale, which is 1.25
  // however that appears to be a mistake
  SIZE_24: '1.35'
};

export { BODY_SIZES, BORDER_WIDTH, BORDER_WIDTH_FORM_ELEMENT, BORDER_WIDTH_FORM_ELEMENT_ERROR, BORDER_WIDTH_FORM_GROUP_ERROR, BORDER_WIDTH_MOBILE, BORDER_WIDTH_WIDE, BREAKPOINTS, CAPTION_SIZES, FOCUSABLE, FOCUSABLE_FILL, FOCUS_WIDTH, FOCUS_WIDTH_RAW, FONT_SIZE, FONT_STACK, FONT_STACK_PRINT, FONT_STACK_TABULAR, FONT_WEIGHTS, FULL_WIDTH, GUTTER, GUTTER_HALF, GUTTER_ONE_THIRD, GUTTER_PX, HEADING_SIZES, LEVEL_SIZE, LEVEL_TAG, LINE_HEIGHT, MEDIA_QUERIES, MEDIA_QUERY_PRINT, NTA_LIGHT, NTA_LIGHT_TABULAR, RESPONSIVE_0, RESPONSIVE_1, RESPONSIVE_2, RESPONSIVE_3, RESPONSIVE_4, RESPONSIVE_5, RESPONSIVE_6, RESPONSIVE_7, RESPONSIVE_8, RESPONSIVE_9, SITE_WIDTH, SITE_WIDTH_PLUS_GUTTERS, SITE_WIDTH_PX, SPACING, SPACING_MAP, SPACING_MAP_INDEX, SPACING_POINTS, TYPOGRAPHY_SCALE, WIDTHS };
