// @govuk-react/text-area/dist/govuk-react-text-area.esm.js
import * as React from 'react';
import styled from 'styled-components';
import { BLACK, YELLOW, ERROR_COLOUR } from 'govuk-colours';
import { NTA_LIGHT, FONT_SIZE, LINE_HEIGHT, MEDIA_QUERIES } from '@govuk-react/constants';
import Label from '@govuk-react/label';
import LabelText from '@govuk-react/label-text';
import ErrorText from '@govuk-react/error-text';
import HintText from '@govuk-react/hint-text';

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};
  var target = _objectWithoutPropertiesLoose(source, excluded);
  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

var _excluded = ["children", "hint", "meta", "input"];

var _styled$withConfig;
var TextAreaField = styled('textarea').withConfig({
  displayName: "src__TextAreaField",
  componentId: "sc-z0ua7p-0"
})((_styled$withConfig = {
  boxSizing: 'border-box',
  fontFamily: NTA_LIGHT,
  fontWeight: 400,
  textTransform: 'none',
  fontSize: FONT_SIZE.SIZE_16,
  lineHeight: LINE_HEIGHT.SIZE_16
}, _defineProperty(_styled$withConfig, MEDIA_QUERIES.LARGESCREEN, {
  fontSize: FONT_SIZE.SIZE_19,
  lineHeight: LINE_HEIGHT.SIZE_19,
  width: '75%'
}), _defineProperty(_styled$withConfig, "width", '100%'), _defineProperty(_styled$withConfig, "padding", '5px 4px 4px'), _defineProperty(_styled$withConfig, "border", "2px solid ".concat(BLACK)), _defineProperty(_styled$withConfig, ':focus', {
  outline: "3px solid ".concat(YELLOW),
  outlineOffset: 0
}), _styled$withConfig), function (_ref) {
  var error = _ref.error;
  return {
    border: error ? "4px solid ".concat(ERROR_COLOUR) : undefined
  };
});
/**
 * Use the textarea component when you need to let users enter an amount of text that’s longer than a single line.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/text-area
 * - https://design-system.service.gov.uk/components/textarea/
 */

var TextArea = function TextArea(_ref2) {
  var children = _ref2.children,
      hint = _ref2.hint,
      meta = _ref2.meta,
      input = _ref2.input,
      props = _objectWithoutProperties(_ref2, _excluded);

  return /*#__PURE__*/React.createElement(Label, _extends({
    error: meta.touched && !!meta.error
  }, props), /*#__PURE__*/React.createElement(LabelText, null, children), hint && /*#__PURE__*/React.createElement(HintText, null, hint), meta.touched && meta.error && /*#__PURE__*/React.createElement(ErrorText, null, meta.error), /*#__PURE__*/React.createElement(TextAreaField, _extends({
    rows: 5,
    error: meta.touched && !!meta.error
  }, input)));
};
TextArea.defaultProps = {
  hint: undefined,
  input: {},
  meta: {}
};
TextArea.displayName = 'TextArea';

export { TextArea, TextAreaField, TextArea as default };
