// @govuk-react/page/dist/govuk-react-page.esm.js
import * as React from 'react';
import SkipLink from '@govuk-react/skip-link';
import TopNav from '@govuk-react/top-nav';
import styled from 'styled-components';
import { spacing } from '@govuk-react/lib';
import { GUTTER, MEDIA_QUERIES, SITE_WIDTH, GUTTER_HALF } from '@govuk-react/constants';

// tracking https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/objects/_main-wrapper.scss
var Main = styled('main').withConfig({
  displayName: "main__Main",
  componentId: "sc-hiygwk-0"
})(spacing.withWhiteSpace({
  padding: {
    size: 6,
    direction: ['top', 'bottom']
  }
}), {
  display: 'block'
});

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

var _styled$withConfig;
var WidthContainer = styled('div').withConfig({
  displayName: "width-container__WidthContainer",
  componentId: "sc-paw03d-0"
})((_styled$withConfig = {
  maxWidth: SITE_WIDTH,
  margin: "0 ".concat(GUTTER_HALF)
}, _defineProperty(_styled$withConfig, MEDIA_QUERIES.TABLET, {
  margin: "0 ".concat(GUTTER)
}), _defineProperty(_styled$withConfig, MEDIA_QUERIES.MAX, {
  margin: '0 auto'
}), _styled$withConfig), spacing.withWhiteSpace());

/**
 * ### References
 *
 * - https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/objects/_main-wrapper.scss
 * - https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/objects/_width-container.scss
 */
/**
 * Use this template to keep your pages consistent with the rest of GOV.UK.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/page
 * - https://design-system.service.gov.uk/styles/page-template/
 * - https://design-system.service.gov.uk/styles/layout/#page-wrappers
 */

var Page = function Page(_ref) {
  var header = _ref.header,
      footer = _ref.footer,
      children = _ref.children,
      id = _ref.id,
      beforeChildren = _ref.beforeChildren,
      MainComponent = _ref.main,
      Container = _ref.container;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SkipLink, {
    href: "#".concat(id)
  }), header, /*#__PURE__*/React.createElement(Container, null, beforeChildren, /*#__PURE__*/React.createElement(MainComponent, {
    id: id
  }, children)), footer);
};
Page.Main = Main;
Page.WidthContainer = WidthContainer;
Page.defaultProps = {
  children: undefined,
  id: 'content',
  header: /*#__PURE__*/React.createElement(TopNav, null),
  footer: undefined,
  main: Page.Main,
  beforeChildren: undefined,
  container: Page.WidthContainer
};
Page.displayName = 'Page';

export { Page, Page as default };
