// @govuk-react/select/dist/govuk-react-select.esm.js
import * as React from 'react';
import styled from 'styled-components';
import { YELLOW, ERROR_COLOUR, BLACK } from 'govuk-colours';
import { MEDIA_QUERIES } from '@govuk-react/constants';
import { typography } from '@govuk-react/lib';
import Label from '@govuk-react/label';
import LabelText from '@govuk-react/label-text';
import ErrorText from '@govuk-react/error-text';
import HintText from '@govuk-react/hint-text';

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};
  var target = _objectWithoutPropertiesLoose(source, excluded);
  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

var _excluded = ["children", "hint", "label", "meta", "input"];

var _styled$withConfig;
var StyledSelect = styled('select').withConfig({
  displayName: "src__StyledSelect",
  componentId: "sc-sgud4a-0"
})(typography.font({
  size: 19
}), (_styled$withConfig = {
  boxSizing: 'border-box',
  width: '100%',
  height: '33px',
  padding: '5px 4px 4px',
  border: "2px solid ".concat(BLACK)
}, _defineProperty(_styled$withConfig, MEDIA_QUERIES.LARGESCREEN, {
  width: '50%',
  height: '38px'
}), _defineProperty(_styled$withConfig, ':focus', {
  outline: "3px solid ".concat(YELLOW),
  outlineOffset: 0
}), _styled$withConfig), function (_ref) {
  var error = _ref.error;
  return {
    border: error ? "4px solid ".concat(ERROR_COLOUR) : undefined
  };
});
/**
 * The select component should only be used as a last resort in public-facing services because research shows that some users find selects very difficult to use.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/select
 * - https://design-system.service.gov.uk/components/select/
 */

var Select = function Select(_ref2) {
  var children = _ref2.children,
      hint = _ref2.hint,
      label = _ref2.label,
      meta = _ref2.meta,
      input = _ref2.input,
      props = _objectWithoutProperties(_ref2, _excluded);

  return /*#__PURE__*/React.createElement(Label, _extends({}, props, {
    error: meta.touched && !!meta.error
  }), /*#__PURE__*/React.createElement(LabelText, null, label), hint && /*#__PURE__*/React.createElement(HintText, null, hint), meta.touched && meta.error && /*#__PURE__*/React.createElement(ErrorText, null, meta.error), /*#__PURE__*/React.createElement(StyledSelect, _extends({
    error: meta.touched && !!meta.error
  }, input), children));
};
Select.defaultProps = {
  hint: undefined,
  errorText: undefined,
  input: {},
  meta: {}
};
Select.displayName = 'Select';
StyledSelect.displayName = 'SelectInput';

export { Select, StyledSelect as SelectInput, Select as default };
