// multi-input-input/es/index.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { Component } from 'react';
import PropTypes from 'prop-types'; // Modifies a component that contains multiple inputs to behave as though it was a single input by:
// - merging values in to an object
// - not calling onBlur when going from field to field within the same component
// - not calling onFocus when going frmo field to field within the same component

export default (function (ComposedComponent) {
  var MultiInputInput =
  /*#__PURE__*/
  function (_Component) {
    _inheritsLoose(MultiInputInput, _Component);

    function MultiInputInput() {
      var _this;

      for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      _this = _Component.call.apply(_Component, [this].concat(args)) || this;

      _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "mergeValueFromEvent", function (e, key) {
        var _extends2;

        return _extends({}, _this.props.value, (_extends2 = {}, _extends2[key] = e.target.value, _extends2));
      });

      _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "handleChange", function (e, key) {
        _this.props.onChange(_this.mergeValueFromEvent(e, key));
      });

      _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "handleBlur", function (e, key) {
        if (!Object.values(_this.inputs).some(function (input) {
          return e.relatedTarget === input;
        })) {
          _this.props.onBlur(_this.mergeValueFromEvent(e, key));
        }
      });

      _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "handleFocus", function (e, key) {
        if (!Object.values(_this.inputs).some(function (input) {
          return e.relatedTarget === input;
        })) {
          _this.props.onFocus(_this.mergeValueFromEvent(e, key));
        }
      });

      return _this;
    }

    var _proto = MultiInputInput.prototype;

    _proto.render = function render() {
      var _this2 = this;

      return React.createElement(ComposedComponent, _extends({
        refs: function refs(inputs) {
          _this2.inputs = inputs;
        }
      }, this.props, this.state, {
        onChange: this.handleChange,
        onBlur: this.handleBlur,
        onFocus: this.handleFocus
      }));
    };

    return MultiInputInput;
  }(Component);

  MultiInputInput.propTypes = {
    value: PropTypes.object,
    onChange: PropTypes.func,
    onBlur: PropTypes.func,
    onFocus: PropTypes.func
  };
  MultiInputInput.defaultProps = {
    value: {},
    onChange: function onChange() {
      return null;
    },
    onBlur: function onBlur() {
      return null;
    },
    onFocus: function onFocus() {
      return null;
    }
  };
  return MultiInputInput;
});
