// @govuk-react/error-summary/dist/govuk-react-error-summary.cjs.dev.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var styled = require('styled-components');
var govukColours = require('govuk-colours');
var heading = require('@govuk-react/heading');
var Paragraph = require('@govuk-react/paragraph');
var UnorderedList = require('@govuk-react/unordered-list');
var Link = require('@govuk-react/link');
var ListItem = require('@govuk-react/list-item');
var constants = require('@govuk-react/constants');
var lib = require('@govuk-react/lib');

function _interopDefault (e) { return e && e.__esModule ? e : { 'default': e }; }

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n["default"] = e;
  return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);
var styled__default = /*#__PURE__*/_interopDefault(styled);
var Paragraph__default = /*#__PURE__*/_interopDefault(Paragraph);
var UnorderedList__default = /*#__PURE__*/_interopDefault(UnorderedList);
var Link__default = /*#__PURE__*/_interopDefault(Link);
var ListItem__default = /*#__PURE__*/_interopDefault(ListItem);

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};
  var target = _objectWithoutPropertiesLoose(source, excluded);
  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

var _excluded = ["onHandleErrorClick", "heading", "description", "errors"];

var _styled$withConfig;
var StyledErrorText = styled__default["default"](Link__default["default"]).withConfig({
  displayName: "src__StyledErrorText",
  componentId: "sc-6d0w2-0"
})((_styled$withConfig = {
  fontFamily: constants.NTA_LIGHT,
  WebkitFontSmoothing: 'antialiased',
  MozOsxFontSmoothing: 'grayscale',
  fontWeight: 700,
  marginBottom: constants.SPACING.SCALE_1,
  textDecoration: 'underline',
  textDecorationSkipInk: 'none',
  textTransform: 'none',
  cursor: 'pointer',
  fontSize: constants.FONT_SIZE.SIZE_16,
  lineHeight: constants.LINE_HEIGHT.SIZE_16,
  ':visited': {
    color: "".concat(govukColours.ERROR_COLOUR)
  }
}, _defineProperty(_styled$withConfig, constants.MEDIA_QUERIES.LARGESCREEN, {
  fontSize: constants.FONT_SIZE.SIZE_19,
  lineHeight: constants.LINE_HEIGHT.SIZE_19
}), _defineProperty(_styled$withConfig, "color", "".concat(govukColours.ERROR_COLOUR)), _defineProperty(_styled$withConfig, "paddingTop", '4px'), _defineProperty(_styled$withConfig, "paddingBottom", '2px'), _styled$withConfig));
var StyledErrorSummary = styled__default["default"]('div').withConfig({
  displayName: "src__StyledErrorSummary",
  componentId: "sc-6d0w2-1"
})(_defineProperty({
  color: govukColours.TEXT_COLOUR,
  padding: constants.RESPONSIVE_4.mobile,
  border: "".concat(constants.BORDER_WIDTH_MOBILE, " solid ").concat(govukColours.ERROR_COLOUR),
  '&:focus': {
    outline: "".concat(constants.FOCUS_WIDTH, " solid ").concat(govukColours.FOCUS_COLOUR),
    outlineOffset: '0'
  }
}, constants.MEDIA_QUERIES.LARGESCREEN, {
  padding: constants.RESPONSIVE_4.tablet,
  border: "".concat(constants.BORDER_WIDTH, " solid ").concat(govukColours.ERROR_COLOUR)
}), lib.spacing.withWhiteSpace({
  marginBottom: 6
}));

/**
 * Use this component at the top of a page to summarise any errors a user has made.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/error-summary
 * - https://design-system.service.gov.uk/components/error-summary/
 */
var ErrorSummary = function ErrorSummary(_ref) {
  var _ref$onHandleErrorCli = _ref.onHandleErrorClick,
      onHandleErrorClick = _ref$onHandleErrorCli === void 0 ? undefined : _ref$onHandleErrorCli,
      _ref$heading = _ref.heading,
      heading$1 = _ref$heading === void 0 ? 'There is a problem' : _ref$heading,
      _ref$description = _ref.description,
      description = _ref$description === void 0 ? undefined : _ref$description,
      _ref$errors = _ref.errors,
      errors = _ref$errors === void 0 ? [] : _ref$errors,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React__namespace.createElement(StyledErrorSummary, _extends({
    tabIndex: -1
  }, props), /*#__PURE__*/React__namespace.createElement(heading.H2, {
    size: "MEDIUM"
  }, heading$1), description && /*#__PURE__*/React__namespace.createElement(Paragraph__default["default"], {
    mb: 3
  }, description), errors.length > 0 && /*#__PURE__*/React__namespace.createElement(UnorderedList__default["default"], {
    mb: 0,
    listStyleType: "none"
  }, errors.map(function (error, index) {
    return /*#__PURE__*/React__namespace.createElement(ListItem__default["default"], {
      key: error.targetName
    }, /*#__PURE__*/React__namespace.createElement(StyledErrorText, {
      tabIndex: 0,
      onClick: function onClick() {
        return onHandleErrorClick === null || onHandleErrorClick === void 0 ? void 0 : onHandleErrorClick(error.targetName);
      }
    }, error.text));
  })));
};
ErrorSummary.defaultProps = {
  description: undefined,
  errors: [],
  heading: 'There is a problem',
  onHandleErrorClick: undefined
};
ErrorSummary.displayName = 'ErrorSummary';

exports.ErrorSummary = ErrorSummary;
exports["default"] = ErrorSummary;
