// @govuk-react/phase-banner/dist/govuk-react-phase-banner.cjs.prod.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var styled = require('styled-components');
var govukColours = require('govuk-colours');
var constants = require('@govuk-react/constants');
var lib = require('@govuk-react/lib');
var Tag = require('@govuk-react/tag');

function _interopDefault (e) { return e && e.__esModule ? e : { 'default': e }; }

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n["default"] = e;
  return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);
var styled__default = /*#__PURE__*/_interopDefault(styled);
var Tag__default = /*#__PURE__*/_interopDefault(Tag);

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};
  var target = _objectWithoutPropertiesLoose(source, excluded);
  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

var _excluded = ["level", "children"];
var StyledBanner = styled__default["default"]('div').withConfig({
  displayName: "src__StyledBanner",
  componentId: "sc-11ccy2-0"
})({
  paddingTop: constants.SPACING_POINTS[2],
  paddingBottom: constants.SPACING_POINTS[2],
  borderBottom: "1px solid ".concat(govukColours.BORDER_COLOUR)
}, lib.spacing.withWhiteSpace());
var BannerContent = styled__default["default"]('p').withConfig({
  displayName: "src__BannerContent",
  componentId: "sc-11ccy2-1"
})(lib.typography.font({
  size: 16
}), lib.typography.textColour, _defineProperty({
  display: 'table',
  margin: 0
}, String(Tag__default["default"]), {
  marginRight: constants.SPACING_POINTS[2]
}));
var BannerText = styled__default["default"]('span').withConfig({
  displayName: "src__BannerText",
  componentId: "sc-11ccy2-2"
})({
  display: 'table-cell',
  verticalAlign: 'baseline'
});
/**
 * Use the phase banner component to show users your service is still being worked on.
 *
 * - https://govuk-react.github.io/govuk-react/?path=/docs/phase-banner
 * - https://design-system.service.gov.uk/components/phase-banner/
 */

var PhaseBanner = function PhaseBanner(_ref) {
  var level = _ref.level,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React__namespace.createElement(StyledBanner, props, /*#__PURE__*/React__namespace.createElement(BannerContent, null, /*#__PURE__*/React__namespace.createElement(Tag__default["default"], null, level), /*#__PURE__*/React__namespace.createElement(BannerText, null, children)));
};
PhaseBanner.displayName = 'PhaseBanner';

exports.PhaseBanner = PhaseBanner;
exports["default"] = PhaseBanner;
